apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/service/DexService.java

```diff
@@ -925,6 +930,53 @@ public EthDepositsWithOffset getUserFilledOrders(String user, long offset, long
         return dexSmartContractService.getUserFilledOrders(user, offset, limit);
     }
 
+    //TODO implement some abstract function to download all data.
+    public List<EthDepositInfo> getAllUserFilledOrders(String user) throws AplException.ExecutiveProcessException {
+        List<EthDepositInfo> ethDepositInfos = new ArrayList<>();
+        int offset = 0;
+        int limit = 100;
+        boolean isContinue = true;
+
+        while (isContinue) {
+            EthDepositsWithOffset ethDepositsWithOffset = dexSmartContractService.getUserFilledOrders(user, offset, limit);
+            if (ethDepositsWithOffset == null) {
```

Comment:
Looks like its not possible to get null here

Reply:
changed on for function